### PR TITLE
Implement minSize and maxSize api

### LINKED ISSUE #28: Add minSize(), maxSize() api to all types
The demand comes from eth2 spec https://github.com/ethereum/eth2.0-specs/pull/1800/files

### PR DESCRIPTION
**Pull request by tuyennhv**
## Goal
+ resolves #28

### CODE REVIEW

File: src/backings/structural/abstract.ts [hunk 1]
```diff
@@ -71,4 +71,10 @@ export class StructuralHandler<T extends object> {
   toJson(value: T, options?: IJsonOptions): Json {
     throw new Error("Not implemented");
   }
+  maxSize(): number {
```
Comment (wemeetagain) [inline src/backings/structural/abstract.ts:74]: can you move these methods to the type objects, eg: move from `src/backings/structural` to `src/types/composite` and `src/types/basic`? These methods would be declared right next to the `size` method for each type

Comment (tuyennhv) [reply src/backings/structural/abstract.ts]: I moved minSize()/maxSize() next to size(). Everything started from `src/type` but for composite, it delegates its implementation to structural backing, just like `size()`

File: src/types/composite/abstract.ts [hunk 2]
```diff
@@ -83,6 +83,20 @@ export class CompositeType<T extends object> {
     }
   }
 
+  /**
+   * Maximal serialized byte length
+   */
+  maxSize(): number {
+    return this.structural.maxSize();
```
Comment (tuyennhv) [inline src/types/composite/abstract.ts:90]: this is the entry point where consumers should call.

**Comment by tuyennhv** [comment]
Got same result like https://gist.github.com/protolambda/db75c7faa1e94f2464787a480e5d613e

Will create unit tests in lodestar to make sure it's correct.

**Pull request merged**